caluma/extensions/events/work_item.py

```diff
@@ -169,22 +169,37 @@ def send_rejection_mail(sender, work_item, user, **kwargs):
     _send_work_item_mail(work_item)
 
 
-@on(post_create_work_item, raise_exception=True)
 @on(post_reopen_case, raise_exception=True)
+@filter_events(lambda case: case.workflow.slug == "document-review")
 @transaction.atomic
-def set_case_status_post_create_work_item_reopen_case(
+def set_case_status_post_reopen_case(
     sender,
+    case,
     user,
-    work_item=None,
-    case=None,
+    work_items,
     **kwargs,
 ):
-    work_item = (
-        work_item if work_item else case.work_items.filter(status="ready").first()
-    )
+    work_item = case.work_items.filter(status="ready").first()
```

Comment:
I'd probably try to use a constant here

```suggestion
    work_item = case.work_items.filter(status=WorkItem.STATUS_READY).first()
```

Reply:
I'm all here for the nitpicks ;)
You are correct and it's fixed :+1: